### ISSUE TITLE
Auto-join Colony with Join URL

### ISSUE THREAD

**Issue by arrenv** [JoinColony/colonyDapp#3403]
### Story

As a user I want to be able to easily join a Colony. It is confusing as a user not knowing that I have to click the `Join` button again and I might forget to do so. (Courtesy of @chinins)

### Description

Create a URL and handle it so that it will attempt to join the user to the colony on load, or if not connected, once they have connected. If the user needs to create a username, then they will be redirected back to the page with the join URL to auto-join.

We can create a new join route `/colony/{COLONY_NAME}/join`

We can also replace the invite link (https://github.com/JoinColony/colonyDapp/issues/3382) as a part of this project with this join URL.

**Comment by ArmandoGraterol** [comment]
@arrenv is this ready to be implemented? It's not clear to me how the process would look like when the user attempts to join the colony via the link

**Comment by arrenv** [comment]
@ArmandoGraterol I don't think it will really change the look of the experience, it will just attempt to automate it.

1. So, the user visits this URL `/colony/{COLONY_NAME}/join`
2. If their wallet is connected, it will attempt to join them
3. If their wallet is not connected they will be directed to create a username
4. Once they have connected their wallet, they will be directed back to `/colony/{COLONY_NAME}/join` to attempt an auto join again.

There is however some designs for a modal to create a username as a part of Onboarding 2.0, so maybe we can wait for that. What do you think?

**Comment by dit7ya** [comment]
Can there be an invite page with just a button `You have been invited to join <colonyName>` <Join/>? This UX is more in line with the expectations of users (given this is what many platform's invite feature look like). I find the silent join action without any visual confirmation a bit odd.

**Comment by arrenv** [comment]
Yeah, it is a good point. Having it auto join you may be a bit off putting. Although, it is kind of how Discord works. In the Onboarding 2.0 there is a dialog. So, maybe this should only be after clicking the join and creating a username.
